Revit_Core_Engine/Convert/Physical/ToRevit/Rebar.cs

```diff
@@ -0,0 +1,125 @@
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Structure;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Physical.Reinforcement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****            Interface methods              ****/
+        /***************************************************/
+
+        public static Rebar IToRevitRebar(this IReinforcingBar reinforcement, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar(reinforcement as dynamic, document, settings, refObjects);
+        }
+
+
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static Rebar ToRevitRebar(this PrimaryReinforcingBar bar, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar((IReinforcingBar)bar, document, settings, refObjects);
+        }
+
+        /***************************************************/
+
+        public static Rebar ToRevitRebar(this Stirrup stirrup, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar((IReinforcingBar)stirrup, document, settings, refObjects);
+        }
+
+
+        /***************************************************/
+        /****              Private methods              ****/
+        /***************************************************/
+
+        private static Rebar ToRevitRebar(this IReinforcingBar bar, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (bar == null || document == null)
+                return null;
+
+            Rebar rebar = refObjects.GetValue<Rebar>(document, bar.BHoM_Guid);
+            if (rebar != null)
+                return rebar;
+
+            settings = settings.DefaultIfNull();
+
+            //getting bar type
+            string barTypeName = ((int)(bar.Diameter * 1000)).ToString();
+            RebarBarType barType = new FilteredElementCollector(document).OfClass(typeof(RebarBarType)).Where(x => x.Name == barTypeName).FirstOrDefault() as RebarBarType;
+            if (barType == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Revit project does not contain adequate rebar family for one or more rebars.");
```

Comment:
I would opt for describing what does _adequate_ mean here (at least to mention the family names here).